Please remove (#396)
3mdeb/news-and-ideas: blog/content/post/2021-02-23-OSF_vs_OCP_OSF.md

or at least documented well enough for an open-source implementation to be
written. The key OSF components have to be open-source, and although it is
allowed to include some closed-source ones *(aka binary blobs)* to OSF - there
should be a valid justification for them to be approved, such as "containing
silicon IP": i.e. a blob has been provided by your SV (Silicon Vendor) like
Intel to set up their chip during boot, which is needed for your device to run.
And all the documentation should be good enough and distributable without NDAs.

![OSF features](/img/OSF_features.png)

The diagram above shows 6 types of OSF features: the fundamental features to the
more advanced ones are depicted from left to right and 1st to 3rd are essential.

1. Chip Initialization and Integration: the firmware components for the initial
setup of the motherboard's key chips - like the chips themselves - are usually
provided by the Silicon Vendors, sometimes in a binary-only form *(without the*
*source code)* like Intel FSP and later AMD AGESA. Some binary-only components,
depending on their complexity, could be reverse-engineered by us at [3mdeb][7] to

> create a higher quality opensource replacement with less bugs and more features.

Reviewer: @mikebdp2 this reply on Reddit show other paths: https://www.reddit.com/r/3mdeb/comments/lf9z0v/qa_post/gn0dj81/?utm_source=reddit&utm_medium=web2x&context=3

Author: @pietrushnic Done, added more information of 3mdeb services to this place.

2. Board Customization: in addition to the chip initialization, there should
be a board-specific setup for the things like GPIO, PCIe bifurcation and BMC
interface. In a coreboot, these could be configured at the mainboard files.

3. System Table Generation: system tables like SMBIOS and ACPI tables have to

> be generated by OSF to provide access to various management features for an OS.

Reviewer: @mikebdp2 what is the meaning of OSF here?

Author: @pietrushnic While writing I meant the open-source but any firmware has to do this, - and anyways the discrepancy between the Open-Source Firmware and Open System Firmware terms should go away after March 2021 _(although not instantly, since there still would be the older OCP-certified products the firmware of which wasn't open-source)_. So I don't think I should expand the OSF term at every mention - otherwise this blog post could become too wordy.

---
title: "Open Compute Open System Firmware (OCP OSF) and its' importance"

> abstract: "Together with Piotr Król let's figure out what is OCP OSF,

Reviewer: ```suggestion
abstract: "Let's figure out what is OCP OSF,
```

>           see how it relates to the open-source firmware - and why it is

Reviewer: ```suggestion
          see how it relates to the Open Source Firmware (OSF) - and why it is
```

  - Firmware
  - Miscellaneous

---

The [Open Compute Project (OCP)][1] is a rapidly growing community with a mission
to design and enable the delivery of the most efficient server, storage and
data-centre hardware designs available for scalable computing. OCP has been
[embraced][2] by the top tier tech giants: IBM, ARM, Google, Facebook, Alibaba,
Intel, Nvidia and even Microsoft - are already the OCP Platinum members. And
over 200 companies have jumped aboard, of course not for the sake of "one more
certification": their customers want to have complete control over their
hardware and its' low-level software (firmware): having the source helps to
prolong their lifecycle by expanding the capabilities, to fix the bugs for
reducing the long-term maintenance fees, and to do their own security audit.
And these customers are willing to pay an extra for the OCP-certified products.

The OCP Open System Firmware (OSF) project is aiming to allow system firmware
like [BIOS][3] and [BMC][4] to be modified and their source code shared with
the owners of OCP-certified hardware. From March 2021, supporting OSF is
mandatory for servers with OCP badging. The ecosystem partners have decided to

> implement such open-source firmware as Coreboot for their OSF architecture.

Reviewer: ```suggestion
implement such open-source firmware as coreboot for their OSF architecture.
```

Just being open-source - by itself is not enough for the firmware to meet the
OCP certification: the extra requirements are mentioned in a [OSF Checklist][5].
The source code should be easily buildable by a native (ideally the open-source)
toolchain and the flashing utility should be either open-source like [flashrom][6]
or at least documented well enough for an open-source implementation to be
written. The key OSF components have to be open-source, and although it is
allowed to include some closed-source ones (aka binary blobs) to OSF - there
should be a valid justification for them to be approved, such as "containing
silicon IP": i.e. a blob has been provided by your SV (Silicon Vendor) like
Intel to set up their chip during boot, which is needed for your device to run.
And all the documentation should be good enough and distributable without NDAs.

As you can expect, even just creating the open-source firmware for your product
is a great challenge that only a few companies in the world can complete, let
alone to maintain it in good shape and to meet all the OCP OSF certification
requirements to enter this new [rapidly growing market][7]. However: we at [3mdeb][8] -

> being the [licensed provider][9] for quality Coreboot consulting services and

Reviewer: ```suggestion
being the [licensed provider][9] for quality coreboot consulting services and
```

> Independent BIOS Vendor (IBV) for [Dasharo][10] coreboot-based firmware - are surely

Reviewer: ```suggestion
Open Source Firmware Vendor (OSFV) relying on [Dasharo][10] to deliver scalable, modular, easy to combine Open Source BIOS, UEFI, and Firmware solutions - are surely
```
